SimCalorimetry/HcalTrigPrimProducers/python/hcaltpdigi_cfi.py

```diff
@@ -24,6 +56,8 @@
     numberOfPresamples = cms.int32(2),
     numberOfSamplesHF = cms.int32(4),
     numberOfPresamplesHF = cms.int32(2),
+    numberOfPresamplesHBQIE11 = cms.int32(0),
+    numberOfPresamplesHEQIE11 = cms.int32(0),
```

Comment:
I see, I am "overloading" the use of npresamples a little bit. I am using npresamples for internal logic to know how many of the weights in the vector are applied to presamples in the filter. For example, with a filter that subtracts w*SOI-1 from SOI, the weight vector would be `[-w, 1.0]` and then I would specify `npresamples = 1`. Conversely, for the current two time sample scheme, with no subtraction, the weight vector is `[1.0, 1.0]` with `npresamples = 0`.

If we always wanted to have two presamples, the above examples could be modified so that the weights vectors would be:
`[0.0, -w, 1.0]` and `[0.0, 0.0, 1.0, 1.0]`, respectively. This seems reasonable to me, assuming I understand the historical use of presamples here.

Reply:
@abdoulline, @christopheralanwest,

Is there a preference between the two scenarios I describe above? I would assume it is best to preserve the "historic" meaning of npresamples here and modify things for the latter scenario(?)

Reply:
Alright, thinking some more, I will decouple notion of number of weighted presamples _in the filter_ from simply the number of presamples that would get stored.

Reply:
With 167641d, I have renamed the two variables to make it more clear that they pertain to the FIR filter. Likewise, the nominal 4 samples and 2 presamples are always used to define an HBHE TP in the end, as has been done before.